docker/Dockerfile

```diff
@@ -16,7 +16,7 @@
 # under the License.
 #
 
-ARG BASE_IMAGE="alpine:3.16"
+ARG BASE_IMAGE="alpine:3.17"
```

Comment:
I think we could use 3.18 if we are to upgrade

Reply:
good idea, https://github.com/apache/nifi-minifi-cpp/pull/1624/commits/c32c9828925057c9967555612a4cac92b3b81e80

Reply:
This caused a lot of tests to fail, so I've reverted it. Will investigate later, but probably in a separate PR